nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -2430,21 +2467,21 @@ impl<'a> Planner<'a> {
     fn can_zone_be_shut_down_safely(
         &self,
         zone: &BlueprintZoneConfig,
-        report: &mut PlanningZoneUpdatesStepReport,
-    ) -> bool {
+    ) -> (bool, BTreeMap<OmicronZoneUuid, ZoneUnsafeToShutdown>) {
```

Comment:
This looks like it always returns either `(true, an_empty_map)` or `(false, a_map_with_exactly_one_entry)` - is that right? If so, we could probably simplify this down to returning `Option<ZoneUnsafeToShutdown>` (where `None` means "it's safe" and `Some(reason)` means "it's unsafe and here's the reason why"). Although probably at that point we should rename the method to something that doesn't sound like it returns a boolean - maybe something like `reason_zone_cannot_be_shut_down_safely()` although hopefully not exactly that because it's super long.